evidence/pool.go

```diff
@@ -205,69 +215,39 @@ func (evpool *Pool) AddEvidence(evidence types.Evidence) error {
 			return fmt.Errorf("failed to verify %v: %w", ev, err)
 		}
 
+		evpool.logger.Info("Hi")
 		// For potential amnesia evidence, if this node is indicted it shall retrieve a polc
-		// to form AmensiaEvidence
-		if pe, ok := ev.(types.PotentialAmnesiaEvidence); ok {
-			var (
-				height = pe.Height()
-				exists = false
-				polc   types.ProofOfLockChange
-			)
-			pe.HeightStamp = evpool.State().LastBlockHeight
-
-			// a) first try to find a corresponding polc
-			for round := pe.VoteB.Round; round > pe.VoteA.Round; round-- {
-				polc, err = evpool.RetrievePOLC(height, round)
-				if err != nil {
-					evpool.logger.Error("Failed to retrieve polc for potential amnesia evidence", "err", err, "pae", pe.String())
-					continue
-				}
-				if err == nil && !polc.IsAbsent() {
-					// we should not need to verify it if both the polc and potential amnesia evidence have already
-					// been verified. We replace the potential amnesia evidence.
-					ae := types.MakeAmnesiaEvidence(pe, polc)
-					err := evpool.AddEvidence(ae)
-					if err != nil {
-						evpool.logger.Error("Failed to create amnesia evidence from potential amnesia evidence", "err", err)
-						// revert back to processing potential amnesia evidence
-						exists = false
-					} else {
-						evpool.logger.Info("Formed amnesia evidence from own polc", "amnesiaEvidence", ae)
-					}
-					break
-				}
+		// to form AmensiaEvidence else start the trial period for the piece of evidence
+		if pe, ok := ev.(*types.PotentialAmnesiaEvidence); ok {
+			if err := evpool.handleInboundPotentialAmnesiaEvidence(pe); err != nil {
+				return err
 			}
-
-			// b) check if amnesia evidence can be made now or if we need to enact the trial period
-			if !exists && pe.Primed(1, pe.HeightStamp) {
-				err := evpool.AddEvidence(types.MakeAmnesiaEvidence(pe, types.EmptyPOLC()))
-				if err != nil {
-					return err
-				}
-			} else if !exists && evpool.State().LastBlockHeight+evpool.State().ConsensusParams.Evidence.ProofTrialPeriod <
-				pe.Height()+evpool.State().ConsensusParams.Evidence.MaxAgeNumBlocks {
-				// if we can't find a proof of lock change and we know that the trial period will finish before the
-				// evidence has expired, then we commence the trial period by saving it in the awaiting bucket
-				pbe, err := types.EvidenceToProto(pe)
-				if err != nil {
-					return err
-				}
-				evBytes, err := pbe.Marshal()
-				if err != nil {
-					return err
+			continue
+		} else if ae, ok := ev.(*types.AmnesiaEvidence); ok {
+			evpool.logger.Info("Hi")
```

Comment:
???

Reply:
Okay lol this was all part of my debugging which I thought I had removed afterwards but turns out I was pretty slack